### PR TITLE
Implement suspend catch for ApplicativeError

### LINKED ISSUE #1732: Add suspend version of catch to `ApplicativeError`
```
suspend fun <A> effectCatch(fe: (Throwable) -> E, f: suspend () -> A): Kind<F, A> = 
  try { f() } catch (t: Throwable) { raiseError(fe(t)) }

suspend fun <A> Applicative<F, Throwable>.effectCatch(f: suspend () -> A): Kind<F, A> = 
  try { f() } catch (t: Throwable) { raiseError(t) }
```

### PR DESCRIPTION
**Pull request by lenguyenthanh**
Close #1732

### CODE REVIEW

File: modules/core/arrow-test/src/main/kotlin/arrow/test/laws/ApplicativeErrorLaws.kt [hunk 1]
```diff
@@ -13,6 +13,7 @@ import arrow.typeclasses.ApplicativeError
 import arrow.typeclasses.Eq
 import io.kotlintest.properties.Gen
 import io.kotlintest.properties.forAll
+import kotlinx.coroutines.runBlocking
```
Comment (pakoito) [inline modules/core/arrow-test/src/main/kotlin/arrow/test/laws/ApplicativeErrorLaws.kt]: Could you please use `IO.effect { }.unsafeRunSync()` instead? :D Dogfood our own lib hahah

**Comment by pakoito** [comment]
90% there and tests passing :D

File: modules/core/arrow-test/src/main/kotlin/arrow/test/laws/ApplicativeErrorLaws.kt (hunk 1 above)
Comment (lenguyenthanh) [reply modules/core/arrow-test/src/main/kotlin/arrow/test/laws/ApplicativeErrorLaws.kt]: Fixed that with the new commit. Good to know this tip.

**Comment by lenguyenthanh** [comment]
I definitely need to do that. I'm quite bad at writing though.

**Comment by pakoito** [comment]
Great chance to practice without pressure then ;)

**Pull request merged**